SpockOpaleye/src/TenantApi.hs

```diff
@@ -15,38 +15,36 @@ module TenantApi
   , deactivateTenant
   ) where
 
-import           ApiBase
+import           AppCore
 import           Control.Arrow
 import           Control.Lens
 import           Control.Monad.Reader
 import           Data.Maybe
 import           Data.Text
-import           DataTypes
 import           GHC.Int
 import           Opaleye
-import           OpaleyeDef
 import           Prelude                    hiding (id)
 import           RoleApi
 import           UserApi
 
-createTenant :: TenantIncoming -> AppM (Auditable Tenant)
+createTenant :: TenantIncoming -> AppM Tenant
 createTenant tenant = do
-  auditable <$> createRow tenantTable tenant
+  createRow tenantTable tenant
 
-activateTenant :: (Auditable Tenant) -> AppM (Auditable Tenant)
+activateTenant :: Tenant -> AppM Tenant
```

Comment:
should take the tenant activation key and validated it first. Or is there a different API for that?